src/main/java/com/uber/cadence/worker/WorkerFactory.java

```diff
@@ -287,6 +289,16 @@ public synchronized void shutdownNow() {
     }
   }
 
+  public CompletableFuture<Boolean> isHealthy() {
```

Comment:
Does it mean that the application will need to call this periodically?

Reply:
Yes. Our clients call this as part of their k8s health check endpoints

Reply:
I see. It's better to have scheduledExecutionService to do the work automatically. This is a nice improvement, but many users don't need to understand this.